vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SpaceOverviewManager.java

```diff
@@ -0,0 +1,298 @@
+package edu.asu.diging.vspace.core.services.impl;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import javax.transaction.Transactional;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import edu.asu.diging.vspace.core.data.ModuleRepository;
+import edu.asu.diging.vspace.core.data.SpaceRepository;
+import edu.asu.diging.vspace.core.data.display.ModuleLinkDisplayRepository;
+import edu.asu.diging.vspace.core.data.display.SpaceLinkDisplayRepository;
+import edu.asu.diging.vspace.core.model.IVSpaceElement;
+import edu.asu.diging.vspace.core.model.display.impl.ModuleLinkDisplay;
+import edu.asu.diging.vspace.core.model.display.impl.SpaceLinkDisplay;
+import edu.asu.diging.vspace.core.model.impl.Module;
+import edu.asu.diging.vspace.core.model.impl.Space;
+import edu.asu.diging.vspace.core.model.impl.SpaceStatus;
+import edu.asu.diging.vspace.core.services.ISpaceOverviewManager;
+import edu.asu.diging.vspace.web.api.ImageApiController;
+import edu.asu.diging.vspace.web.staff.ModuleController;
+import edu.asu.diging.vspace.web.staff.SpaceController;
+
+@Service
+public class SpaceOverviewManager implements ISpaceOverviewManager {
+
+    @Autowired
+    private SpaceRepository spaceRepo;
+
+    @Autowired
+    private ModuleRepository moduleRepo;
+
+    @Autowired
+    private SpaceLinkDisplayRepository spaceLinkDisplayRepository;
+
+    @Autowired
+    private ModuleLinkDisplayRepository moduleLinkDisplayRepository;
+
+    /**
+     * To map each space to all linked space
+     */
+    @Override
+    public Map<String, List<ModuleLinkDisplay>> getSpaceToModuleLinks() {
+
+        Map<String, List<ModuleLinkDisplay>> spaceToModuleLinksMap = new HashMap<>();
+        Iterable<ModuleLinkDisplay> spaceToModuleLinksList = moduleLinkDisplayRepository.findAll();
+
+        for (ModuleLinkDisplay link : spaceToModuleLinksList) {
+            String spaceId = null;
+            if (link.getLink() != null && link.getLink().getSpace() != null
+                    && link.getLink().getSpace().getId() != null) {
```

Comment:
`getId` should never return null, since all ModuleLinkDisplay are coming directly from the database hence all objects should have an id.